packages/core/pluggableElementTypes/renderers/ComparativeServerSideRendererType.ts

```diff
@@ -93,7 +95,12 @@ export default class ComparativeServerSideRenderer extends ServerSideRenderer {
   }
 
   async getFeatures(renderArgs: any) {
-    const { dataAdapter, signal } = renderArgs
+    const { signal, sessionId, adapterConfig } = renderArgs
+    const { dataAdapter } = await getAdapter(
```

Comment:
why did this change to be getAdapter instead of receiving it directly in renderArgs?

Reply:
changed so we can pass pluginManager in the adapter, when it was receiving directly in renderArgs we had a circular reference where pluginManager had renderArgs which had adapterConfig which had pluginManager that was getting stringified and threw an error